pandas/io/tests/test_gbq.py

```diff
@@ -246,6 +347,24 @@ def tearDown(self):
         # executed.
         pass
 
+    def test_should_read_as_service_account_with_key_path(self):
+        if not PRIVATE_KEY_JSON_PATH:
+            raise nose.SkipTest("Cannot run integration tests without a "
```

Comment:
make this a function at the top of the file
``_skip_if_no_private_key()``, and just call it where you need